plugins/destination/s3/client/spec/spec.go

```diff
@@ -61,6 +63,12 @@ type Spec struct {
 	// **Note**: if you want to use path-style addressing, i.e., `https://s3.amazonaws.com/BUCKET/KEY`, `use_path_style` should be enabled, too.
 	Endpoint string `json:"endpoint,omitempty"  jsonschema:"default="`
 
+	// SSE KMS Key ID appened to S3 API calls header. Used in conjuction with server_side_encryption.
+	SSEKMSKeyId string `json:"sse_kms_key_id,omitempty" jsonschema:"default="`
+
+	//Server Side Encryption header which declares encryption type in S3 API calls header: x-amz-server-side-encryption.
+	ServerSideEncryption types.ServerSideEncryption `json:"server_side_encryption,omitempty" jsonschema:"default="`
```

Comment:
```suggestion
	ServerSideEncryption types.ServerSideEncryption `json:"server_side_encryption,omitempty" jsonschema:"enum=AES256,enum=aws:kms,enum=aws:kms:dsse"`
```

Reply:
Also need to run `make gen-spec-schema` to update the JSON schema

Reply:
I believe you are correct, the `read.go` shouldn't need changing.